### PR TITLE
fix(healthmanager)_: extract subscriber logic from RPC Health Manager

### LINKED ISSUE #6139: Flaky test: `TestDelayedChainUpdate`
```
=== RUN   TestBlockchainHealthManagerSuite/TestDelayedChainUpdate
    blockchain_health_manager_test.go:48: 
        	Error Trace:	/Users/igorsirotin/Repositories/Status/status-go/healthmanager/blockchain_health_manager_test.go:48
        	            				/Users/igorsirotin/Repositories/Status/status-go/healthmanager/blockchain_health_manager_test.go:42
        	            				/Users/igorsirotin/Repositories/Status/status-go/healthmanager/providers_health_manager_test.go:202
        	Error:      	Not equal: 
        	            	expected: "down"
        	            	actual  : "up"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-(rpcstatus.StatusType) (len=4) "down"
        	            	+(rpcstatus.StatusType) (len=2) "up"
        	            	 
        	Test:       	TestBlockchainHealthManagerSuite/TestDelayedChainUpdate
        	Messages:   	Expected blockchain status to be down
--- FAIL: TestBlockchainHealthManagerSuite/TestDelayedChainUpdate (0.00s)
```

### PR DESCRIPTION
**Pull request by friofry**
* Subscription common logic is extracted to a separate type.
* Fix race where go routing extracts value from sync.map and then another go routine calls unsubscribe and closes the channel before the first go routing tries to write to the channel.
* moved TestInterleavedChainStatusChanges TestDelayedChainUpdate to the correct file
* Renamed test suites with duplicate names

closes #6139

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -14,3 +14,4 @@ shell.nix             @status-im/devops
 # Feel free to add yourself for any new packages you implement.
 /cmd/status-backend   @igor-sirotin
 /internal/sentry      @igor-sirotin
+/healthmanager	      @friofry
```
Comment (igor-sirotin) [inline .github/CODEOWNERS:17]: 🔥

File: healthmanager/subscription_manager_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	sm.Emit(ctx)
+
+	// Wait for all goroutines to finish
+	wg.Wait()
+
+	// Verify that all subscribers received the notification
+	for i, received := range notificationReceived {
+		if !received {
+			t.Errorf("Subscriber %d did not receive notification", i)
+		}
```
Comment (igor-sirotin) [inline healthmanager/subscription_manager_test.go]:
```suggestion
        require.Truef(t, received, "Subscriber %d did not receive notification", i)
```

File: healthmanager/subscription_manager_test.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	sm.Unsubscribe(ch)
+
+	// Attempt to send a notification
+	sm.Emit(ctx)
+
+	select {
+	case _, ok := <-ch:
+		if ok {
+			t.Errorf("Unsubscribed channel received notification")
+		}
```
Comment (igor-sirotin) [inline healthmanager/subscription_manager_test.go]:
```suggestion
		require.Falsef(t, ok, "Unsubscribed channel received notification")
```

File: healthmanager/subscription_manager_test.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+	case <-time.After(100 * time.Millisecond):
+		// Expected behavior: no notification should be received
```
Comment (igor-sirotin) [inline healthmanager/subscription_manager_test.go]: Not quite good thing really: [**Prevent flaky tests:** don’t wait for absence of something](https://www.notion.so/Prevent-flaky-tests-2045334180cb4eabbe2151f7005e16ce?pvs=4#e77e978d5a694b2dac7552ef68238913)

Not sure what's the best approach in this case though. But perhaps just checking that the channel doesn't exist in the `SubscriptionManager.subscription` is enough. You have access to the private variable here.

File: healthmanager/subscription_manager_test.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+	sm.Emit(ctx)
+	sm.Emit(ctx)
+
+	// Allow some time for the goroutine to process notifications
+	time.Sleep(100 * time.Millisecond)
+
+	mu.Lock()
+	if received != 1 {
+		t.Errorf("Expected 1 notifications, but received %d", received)
+	}
```
Comment (igor-sirotin) [inline healthmanager/subscription_manager_test.go]:
```suggestion
	require.Equalf(t, 1, received, "Expected 1 notifications, but received %d", received)
```

**Review by igor-sirotin — APPROVED** [review]
Nice, thank you!

File: healthmanager/subscription_manager_test.go (hunk 4 above)
Comment (friofry) [reply healthmanager/subscription_manager_test.go]: fixed
Thanks for pointing out this page!

**Review by saledjenic — APPROVED** [review]

**Pull request merged**